mongo/integration/database_test.go

```diff
@@ -237,18 +237,24 @@ func TestDatabase(t *testing.T) {
 			assert.Nil(mt, err, "expected command %s to contain key 'batchSize'", evt.Command)
 		})
 
-		// The BatchSize option is not honored for ListCollections operations on server version 2.6 due to an
-		// inconsistency in the legacy OP_QUERY code path (GODRIVER-1937).
-		cmdMonitoringMtOpts := mtest.NewOptions().MinServerVersion("3.0")
-		mt.RunOpts("getMore commands are monitored", cmdMonitoringMtOpts, func(mt *mtest.T) {
+		// For server versions below 3.0, we internally execute ListCollections() as a legacy
+		// OP_QUERY against the system.namespaces collection. Command monitoring upconversions
+		// translate this to a "find" command rather than "listCollections".
+		cmdMonitoringCmdName := "listCollections"
+		if mtest.CompareServerVersions(mtest.ServerVersion(), "3.0") < 0 {
+			cmdMonitoringCmdName = "find"
+		}
+		mt.Run("getMore commands are monitored", func(mt *mtest.T) {
 			createCollections(mt, 2)
-			assertGetMoreCommandsAreMonitored(mt, "listCollections", func() (*mongo.Cursor, error) {
+
```

Comment:
nit: unnecessary newline

Reply:
I did miss your comment about re-enabling the previous batch size test as well. I've made that change as well as removed the newlines.